src/averageAges.js

```diff
@@ -56,6 +85,38 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  let ageDiffArr;
+
+  if (onlyWithSon) {
+    const manArr = people.filter(person => person.sex === 'm');
+
+    ageDiffArr = manArr.map(person => {
+      const motherArr = people.filter(el => el.name === person.mother);
+      const mother = motherArr[0];
+
+      if (mother) {
```

Comment:
according to the task rules, you should use only ternary operators instead of if else

Reply:
Not resolved.